### PR TITLE
Implement Bolt command

### LINKED ISSUE #50: Absent command
Implement command to allow users to count the number of times each student is absent.

A lighting bolt would be used to represent number of absences.

### PR DESCRIPTION
**Pull request by kwuunnn**
Allow users to count number of absences per student.

Let's

- Create BoltCommand
- Create Bolt class
- Implement testing for Bolt commands

Fixes #50

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/BoltCommand.java [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+import seedu.address.commons.core.index.Index;
+import seedu.address.commons.util.ToStringBuilder;
+import seedu.address.logic.Messages;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.person.Bolt;
+import seedu.address.model.person.Person;
+
+/**
+ * Adds a bolt to a student in the address book.
```
Comment (evedaktyl) [inline src/main/java/seedu/address/logic/commands/BoltCommand.java:17]: Seems to imply that only one bolt is added, whereas the bolt command allows the adding of multiple bolts at once. Could we rephrase this to represent the command more accurately?

File: src/main/java/seedu/address/logic/commands/BoltCommand.java [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+ * @code bolt 1 b/3
+ */
+public class BoltCommand extends Command {
+
+    public static final String COMMAND_WORD = "bolt";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Adds the number of lightning bolts to "
+            + "the person identified by the index number used in the last person listing. "
+            + "The number of lightning bolts corresponds to the number of absences. "
+            + "Updates the existing absence count by adding the input number.\n"
```
Comment (laksherthanlife) [inline src/main/java/seedu/address/logic/commands/BoltCommand.java:27]: Minor nit but maybe it would be clearer to say something like "Adds the input number to the existing absence count"?

File: src/main/java/seedu/address/logic/commands/BoltCommand.java [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+    public CommandResult execute(Model model) throws CommandException {
+        List<Person> lastShownList = model.getCorrectPersonList(); // get the list of persons
+
+        if (index.getZeroBased() >= lastShownList.size()) { // if index out of range
+            throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+        }
+
+        Person personToEdit = lastShownList.get(index.getZeroBased()); // get Person indexed
+        Bolt boltToEdit = personToEdit.getBolt(); // return the bolts
+        Bolt editedBolt = new Bolt(boltToEdit.numOfBolts + this.bolt.numOfBolts); // new Bolt to be added
```
Comment (laksherthanlife) [inline src/main/java/seedu/address/logic/commands/BoltCommand.java:59]: does this violate tell dont ask? would it be better to instead have a static method in both to handle combining two bolts?

File: src/main/java/seedu/address/logic/commands/BoltCommand.java [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+ */
+public class BoltCommand extends Command {
+
+    public static final String COMMAND_WORD = "bolt";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Adds the number of lightning bolts to "
+            + "the person identified by the index number used in the last person listing. "
+            + "The number of lightning bolts corresponds to the number of absences. "
+            + "Updates the existing absence count by adding the input number.\n"
+            + "Parameters: INDEX (must be a positive integer) "
```
Comment (laksherthanlife) [inline src/main/java/seedu/address/logic/commands/BoltCommand.java:28]: in the `Bolt` class, it accepts a value of 0? Should we change this to be "must be a nonnegative integer"?

File: src/main/java/seedu/address/logic/parser/StarCommandParser.java [hunk 5]
```diff
@@ -24,7 +24,7 @@ public class StarCommandParser implements Parser<StarCommand> {
     public StarCommand parse(String args) throws ParseException {
         requireNonNull(args);
         ArgumentMultimap argMultimap =
-                ArgumentTokenizer.tokenize(args, PREFIX_STAR); // tokenise args by PREFIX_STAR
+                ArgumentTokenizer.tokenize(args, PREFIX_STAR); // tokenize args by PREFIX_STAR
```
Comment (laksherthanlife) [inline src/main/java/seedu/address/logic/parser/StarCommandParser.java:27]: Just to check, do we standardise to American English spelling?

File: src/test/java/seedu/address/logic/commands/BoltCommandTest.java [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+import seedu.address.model.ModelManager;
+import seedu.address.model.UserPrefs;
+import seedu.address.model.person.Bolt;
+import seedu.address.model.person.Person;
+import seedu.address.testutil.PersonBuilder;
+
+/**
+ * Contains integration tests (interaction with the Model) and unit tests for BoltCommand.
+ */
+public class BoltCommandTest {
```
Comment (laksherthanlife) [inline src/test/java/seedu/address/logic/commands/BoltCommandTest.java:26]: Nice test cases!

**Review by laksherthanlife — COMMENTED** [review]
Overall lgtm. Just a few minor nits regarding tell don't ask

File: src/main/java/seedu/address/logic/commands/BoltCommand.java (hunk 4 above)
Comment (evedaktyl) [reply src/main/java/seedu/address/logic/commands/BoltCommand.java:28]: @R-Laksh Perhaps the bolt class accepts 0 so that the bolt field can be initialised as 0, but the command to add bolt should still take a positive integer, otherwise the command seems a bit meaningless

**Review by c-wenlong — COMMENTED** [review]
LGTM!

**Review by evedaktyl — COMMENTED** [review]
LGTM, suggested some changes.

File: src/main/java/seedu/address/logic/parser/StarCommandParser.java (hunk 5 above)
Comment (kwuunnn) [reply src/main/java/seedu/address/logic/parser/StarCommandParser.java:27]: Yeap! I think it was mentioned in the coding standard. Make America great again!
